test/integration/resourcemanager/managedresourcecontroller/managedresource_controller_test.go

```diff
@@ -0,0 +1,222 @@
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: corev1.SchemeGroupVersion.String(),
+						Kind:       "Secret",
+					}, ObjectMeta: metav1.ObjectMeta{
+						Name:      updateTestSecretName,
+						Namespace: namespaceName,
+					}}
+				Expect(testClient.Get(ctx, client.ObjectKeyFromObject(secretForRef), secretForRef)).To(Or(Succeed(), BeNoMatchError()))
+				secretForRef.Data, err = appendSecretData(updateTestSecret, "update-test-secret.yaml", secretForRef.Data)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testClient.Update(ctx, secretForRef)).To(Succeed())
+
+				Eventually(func() error {
+					err := testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)
+					if err != nil {
+						return err
+					}
+					return testClient.Get(ctx, client.ObjectKeyFromObject(updateTestSecret), updateTestSecret)
+				}, time.Minute, 5*time.Second).Should(Succeed())
+
+				Eventually(func() bool {
+					err := testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)
+					condition := getConditionByType(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+					return err == nil && condition != nil && condition.Status == v1beta1.ConditionTrue
+				}, time.Minute, time.Second).Should(BeTrue())
+			})
+
+			It("the managed resource update should fail", func() {
+				err = testClient.Create(ctx, managedResource)
+				Expect(err).To(Or(Succeed(), BeAlreadyExistsError()))
+
+				updateTestSecret = &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      updateTestSecretName,
+						Namespace: namespaceName,
+					}}
+				Expect(testClient.Get(ctx, client.ObjectKeyFromObject(secretForRef), secretForRef)).To(Or(Succeed(), BeNoMatchError()))
+				secretForRef.Data, err = appendSecretData(updateTestSecret, "update-test-secret.yaml", secretForRef.Data)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testClient.Update(ctx, secretForRef)).To(Succeed())
+
+				Eventually(func() bool {
+					err := testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)
+					condition := getConditionByType(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied)
+					return err == nil && condition != nil && condition.Status == v1beta1.ConditionFalse
+				}, time.Minute, time.Second).Should(BeTrue())
+			})
+		})
+	})
+})
+
+func createSecretDataFromObject(obj runtime.Object, key string) (*map[string][]byte, error) {
+	jsonObject, err := json.Marshal(obj)
+	if err != nil {
+		return nil, err
+	}
+	data := make(map[string][]byte)
+	data[key] = jsonObject
+	return &data, nil
+}
+
+func appendSecretData(obj runtime.Object, key string, data map[string][]byte) (map[string][]byte, error) {
+	if _, ok := data[key]; ok {
+		return nil, fmt.Errorf("Key: %s already exists", key)
+	}
+	data[key], err = json.Marshal(obj)
+
+	return data, err
+}
+
+func getConditionByType(conditions []v1beta1.Condition, conditionType v1beta1.ConditionType) *v1beta1.Condition {
```

Comment:
You could check whether this can be replaced by the existing `pkg/apis/core/v1beta1/helper.GetCondition` func.